gear/redcap_error_checks_import/src/python/redcap_error_checks_import_app/visitor.py

```diff
@@ -0,0 +1,135 @@
+"""Module to handle downloading error check CSVs from S3."""
+import logging
+from typing import Any, Dict, List
+
+from inputs.csv_reader import CSVVisitor
+from outputs.errors import (
+    LogErrorWriter,
+    empty_field_error,
+    missing_field_error,
+    unexpected_value_error,
+)
+
+from .utils import ErrorCheckKey
+
+log = logging.getLogger(__name__)
+
+
+class ErrorCheckCSVVisitor(CSVVisitor):
+    """Visitor for an Error Check CSV file."""
+
+    # error_no, do_in_redcap, in_prev_versions,
+    # questions, and any other extra headers ignored
+    REQUIRED_HEADERS = ("error_code", "error_type", "form_name", "packet",
+                        "var_name", "check_type", "test_name", "short_desc",
+                        "full_desc", "test_logic", "comp_forms", "comp_vars")
+
+    ALLOWED_EMPTY_FIELDS = ("comp_forms", "comp_vars")
+
+    def __init__(self, key: ErrorCheckKey,
+                 error_writer: LogErrorWriter) -> None:
+        """Initializer."""
+        self.__key = key
+        self.__error_writer = error_writer
+        self.__validated_error_checks: List[Dict[str, Any]] = []
+
+    @property
+    def error_writer(self) -> LogErrorWriter:
+        """Get the error writer.
+
+        Returns:
+            The LogErrorWriter.
+        """
+        return self.__error_writer
+
+    @property
+    def validated_error_checks(self) -> List[Dict[str, Any]]:
+        """Get the validated error checks.
+
+        Returns:
+            The running list of error checks.
+        """
+        return self.__validated_error_checks
+
+    def visit_header(self, header: List[str]) -> bool:
+        """Adds the header, and asserts all required fields are present.
+
+        Args:
+          header: list of header names
+        Returns:
+          True if the header has all required fields, False otherwise
+        """
+        error = None
+        for h in self.REQUIRED_HEADERS:
```

Comment:
Not that I expect a change, but we should converge on a strategy for checking these. I have some code elsewhere that uses the set.subset operation to check this

Reply:
an example: https://github.com/naccdata/flywheel-gear-extensions/blob/0f4f8c6ae3bf03bbb5f31054af00fb1012594ad3/gear/identifier_provisioning/src/python/identifier_provisioning_app/main.py#L110